File: java_programming_construct/Day5/LeapYear.java
package java_programming_construct.basic_core_java_functional_program;

import java.util.Scanner;

//PROBLEM 2(Leap Year)
public class LeapYear {
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter year: ");
        int year = sc.nextInt();
        if (year < 1000 || year > 9999) {
            System.out.println("Enter a 4 digit year");
        } else if ((year % 400 == 0) || (year % 4 == 0 && year % 100 != 0)) {
            System.out.println(year + " is a Leap Year");
        } else {
            System.out.println(year + " is not a Leap Year");
        }
    }
}

File: web_intro/ProgrammingConstruct/SelectionProblems-If-else/LeapYear.js
// Program to check whether a year is a Leap Year or not

let year = 2024;

// Check if it is a 4-digit year
if (year >= 1000 && year <= 9999) {

    // Leap year condition
    if ((year % 4 === 0 && year % 100 !== 0) || (year % 400 === 0)) {
        console.log(year + " is a Leap Year");
    } else {
        console.log(year + " is NOT a Leap Year");
    }

} else {
    console.log("Please enter a valid 4-digit year");
}